Change API Gin metrics package

rh-hemartin opened the pull request:
<!--  Thanks for sending a pull request!

Please read our contributor guidelines:
https://kubearchive.github.io/kubearchive/main/contributors/guide.html
-->

## Which Issue(s) this Pull Request Resolves
<!-- Please make sure to create an issue you can link to and mention it here
to automatically close it. If this PR covers part of the work, instead of
"Resolves #", use "Related to #"

Usage: `Resolves #<issue number>`, or `Resolves (paste link of issue)`.
-->

Related to #527 527

## Release Note
<!--
If this change has user-visible impact, write a release note in the block
below. Include the string "action required" if additional action is required of
users switching to the new release, for example in case of a breaking change.

Write as if you are speaking to users, not other KubeArchive contributors. If this
change has no user-visible impact, leave the code block as is.
-->

```release-note
NONE
```

## Notes for Reviewers
<!--
Leave notes for the reviewers if you want to point their attention to some
specific place.
-->

* The current instrumentation package for gin does not provide metrics, I'm changing it to one that does.
* The current instrumentation used is deprecated and has no maintainer: https://github.com/open-telemetry/opentelemetry-go-contrib/issues/6190

<!--
Please label this pull request according to what type of issue you are addressing.
For reference on required PR/issue labels, read here:
https://kubearchive.github.io/kubearchive/main/contributors/release.html#_pull_request_labels

Add one of the following labels:
kind/bug
kind/documentation
kind/feature

Optionally add one or more of the following kinds if applicable:
kind/breaking
-->

maruiz93 commented:
Can you please develop the need of why should this change done? Maybe by linking a ticket with the proposed change?

As far as I am aware the previous package `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin` is the "official" one and the proposed one doesn't have details on what differentiates it from the official package.

maruiz93 reviewed (changes requested):
Can you please develop the need of why should this change done? Maybe by linking a ticket with the proposed change?

As far as I am aware the previous package `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin` is the "official" one and the proposed one doesn't have details on what differentiates it from the official package.

rh-hemartin commented:
> Can you please develop the need of why should this change done? Maybe by linking a ticket with the proposed change?
> 
> As far as I am aware the previous package `go.opentelemetry.io/contrib/instrumentation/github.com/gin-gonic/gin/otelgin` is the "official" one and the proposed one doesn't have details on what differentiates it from the official package.

I added some reviewers notes to the PR.

rh-hemartin commented:
Hey @maruiz93 and @skoved I changed something across all tests. We were using `http.NewRequest` with `nil` as body. This caused the Body of the request to be `nil`, which then made the new middleware we are introducing fail because of that. We changed all `http.NewRequest` to `httptest.NewRequest` which does not set `Body` as `nil` if `nil` is passed as body parameter.

I checked that in the real scenario, when you call the API with curl the body is not nil but an empty string. Instead of passing `strings.NewReader("")` to all `http.NewRequest` I think its better to go with the `httptest` package which already accounts for that.

Let me know what you think about this.
